pallets/ajuna-tournament/src/traits.rs

```diff
@@ -0,0 +1,139 @@
+use frame_support::{pallet_prelude::*, PalletId};
+use parity_scale_codec::{Error, Input};
+use sp_runtime::TypeId;
+
+pub const REWARD_TABLE_MAX_LENGTH: u32 = 11;
+
+pub const MAX_PLAYERS: u32 = 10;
+
+pub type TournamentId = u32;
+
+pub type RewardTable = BoundedVec<u8, ConstU32<REWARD_TABLE_MAX_LENGTH>>;
+
+pub type PlayerTable<T> = BoundedVec<T, ConstU32<MAX_PLAYERS>>;
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct TournamentTreasuryAccount<SeasonId> {
+	pub pallet_id: PalletId,
+	pub season_id: SeasonId,
+	pub tournament_id: TournamentId,
+}
+
+type TreasuryAccountEncodec<'a, SeasonId> =
+	(&'a PalletId, &'a [u8; 1], &'a SeasonId, &'a [u8; 1], &'a TournamentId);
+
+type TreasuryAccountDecodec<SeasonId> = (PalletId, [u8; 1], SeasonId, [u8; 1], TournamentId);
+
+impl<SeasonId: Encode> Encode for TournamentTreasuryAccount<SeasonId> {
+	fn encode(&self) -> Vec<u8> {
+		// This codec will fit into the indexers rendering design such that we can
+		// see the treasury accounts as "<pallet_id>/season_id/tournament_id".
+		let data: TreasuryAccountEncodec<SeasonId> =
+			(&self.pallet_id, b"/", &self.season_id, &b"/", &self.tournament_id);
+		data.encode()
+	}
```

Comment:
Hmm, now I am having doubts about using a struct for this. Maybe it is better to just have functions that create the encoded value. It is not good to implement a non-standard parity-scale-codec as all the libraries that autogenerate types, will fail to properly de-/encode it.

I did this to benefit from the auto-implemented `into_account_truncating` if the type satisfies `TypeId`, `Encode`, `Decode`, but I think now that it is actually better to simply write our own `into_account_truncating` replicating the original one, and just derive encode and decode on this type. What do you think?

Reply:
I suggest we simply write a function `into_treasury_account_truncating`. I am not even sure if we ever need the "decode" part.